src/Tuura/Concept/Circuit/Derived.hs

```diff
@@ -82,6 +84,42 @@ dual c = mempty
            invariant = fmap dualInvariant (invariant c)
          }
 
+-- Give the opposite initial state for the chosen signal
+bubbleInitialValue :: Eq a => a -> (a -> InitialValue) -> (a -> InitialValue)
+bubbleInitialValue s f y = if (y == s) then bubbleVal (f y) else f y
+  where
+    bubbleVal (Defined v) = Defined (not v)
+    bubbleVal x = x
+
+-- Perform the inversion for just transitions of the given signal
+invertSignal :: Eq a => a -> Transition a -> Transition a
```

Comment:
There is function `toggle` which is almost the same, so for consistency we should keep `toggle` in the name instead of introducing additional `invert` naming. I think we need something like `toggleSignal`, `toggleSpecific` or similar.